File: backend/app/schemas/notification.py
"""Pydantic v2 schemas for notifications."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import NotificationCategory, NotificationType


class NotificationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    message: str
    type: NotificationType
    category: NotificationCategory
    is_read: bool
    action_url: str | None
    related_entity_type: str | None
    related_entity_id: int | None
    created_at: datetime


class UnreadCount(BaseModel):
    unread: int


class ScanResult(BaseModel):
    created: int

File: frontend/src/types/notification.ts
export interface Notification {
  id: number;
  title: string;
  message: string;
  type: 'info' | 'success' | 'warning' | 'alert' | 'reminder';
  category: string;
  is_read: boolean;
  action_url: string | null;
  related_entity_type: string | null;
  related_entity_id: number | null;
  created_at: string;
}

export interface UnreadCount {
  unread: number;
}

export interface ScanResult {
  created: number;
}
